SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -92,6 +92,15 @@ internal void LoadCache()
         internal void SaveCache()
         {
             var savePathDir = GetCachePath();
+
+            if(File.Exists(savePathDir)) // File exists
+            {
+                if(FileUtils.IsFileLocked(savePathDir))
+                {
+                    return; // File is locked
```

Comment:
Assuming this condition happens, what part of the code is responsible for reattempting to write the file after this call into `SaveCache` was cut early?

Reply:
TechTypePatcher.Patch? Don't get what you mean

Reply:
`TechTypePatcher.Patch` is only calling `cacheManager.SaveCache();` and doesn't appear to be checking for any success/fail response.

Perhaps the `SaveCache` method should return a bool informing if the file was successfully saved.  
So in the event that it isn't, the caller can reattempt it after a short wait.